pos-print/src/main/java/com/clouway/pos/print/adapter/db/PersistentReceiptRepository.kt

```diff
@@ -0,0 +1,136 @@
+package com.clouway.pos.print.adapter.db
+
+import com.clouway.pos.print.core.*
+import com.google.inject.Inject
+import com.google.inject.Provider
+import com.mongodb.client.MongoCollection
+import com.mongodb.client.MongoDatabase
+import com.mongodb.client.model.Filters.eq
+import com.mongodb.client.model.Updates.set
+import org.bson.Document
+import org.slf4j.LoggerFactory
+
+/**
+ * @author Tsvetozar Bonev (tsvetozar.bonev@clouway.com)
+ */
+class PersistentReceiptRepository @Inject constructor(private val database: Provider<MongoDatabase>)
+  : ReceiptRepository, PrintingListener {
+  private val collectionName: String = "receipts"
+
+  override fun register(receipt: Receipt): String {
+    if (receipts().find(eq("receiptId", receipt.receiptId)).firstOrNull() != null)
+      throw ReceiptAlreadyInQueueException()
+
+    val receiptDoc = receipt.toDocument()
+
+    receipts().insertOne(receiptDoc)
+
+    return receipt.receiptId
+  }
+
+  override fun getStatus(receiptId: String): PrintStatus {
+    val receiptDoc = receipts().find(
+      eq("receiptId", receiptId)).firstOrNull()
+      ?: throw ReceiptNotInQueueException()
```

Comment:
why ReceiptNotInQueueException ? 
I think ReceiptNotFoundException  fits better, actually we check the status.
We can have a receipt with status printed, that will not be in the queue for printing, that's why the name of the exception sounds confusing to me.

Reply:
Renamed exceptions to ReceiptNotFoundException and ReceiptAlreadySavedExceptions for clarity.